packages/material-ui/src/ButtonBase/ButtonBase.test.js

```diff
@@ -88,6 +88,27 @@ describe('<ButtonBase />', () => {
       expect(button).to.have.attribute('href', 'https://google.com');
     });
 
+    it('should use custom LinkComponent when provided', () => {
+      const CustomLink = React.forwardRef((props, ref) => {
+        return (
+          <a data-testid="customLink" ref={ref} {...props}>
+            {props.children}
+          </a>
+        );
+      });
+
+      const { container, getByTestId } = render(
+        <ButtonBase LinkComponent={CustomLink} href="https://google.com">
```

Comment:
I think that the test case should use the theme. Without, this prop has no use case, it's actually harmful since developers can use the `component` prop directly.

Reply:
updated tests.